packages/x-charts/src/ChartsHtmlLegend/ChartsLegend.tsx

```diff
@@ -0,0 +1,167 @@
+'use client';
+import * as React from 'react';
+import { styled } from '@mui/material/styles';
+import { PrependKeys } from '@mui/x-internals/types';
+import PropTypes from 'prop-types';
+import { useLegend } from '../hooks/useLegend';
+import { ChartsLegendItem } from './ChartsLegendItem';
+import type { ChartsLegendPlacement, ChartsLegendSlotExtension } from './legend.types';
+import { SeriesLegendItemContext } from './legendContext.types';
+import { ChartsLabelProps } from '../ChartsLabel/ChartsLabel';
+import { ChartsLabelMarkProps } from '../ChartsLabel/ChartsLabelMark';
+import { seriesContextBuilder } from './onClickContextBuilder';
+import { useUtilityClasses, type ChartsLegendClasses } from './chartsLegendClasses';
+import { consumeSlots } from '../internals/consumeSlots';
+
+export interface ChartsLegendProps
+  extends ChartsLegendPlacement,
+    Omit<ChartsLabelProps, 'children'>,
+    PrependKeys<Omit<ChartsLabelMarkProps, 'color'>, 'mark'> {
+  // TODO: should be handled by the `BarChart/LineChart/etc` components
+  // hidden?: boolean
+  /**
+   * Space between two legend items (in px).
+   * @default theme.spacing(2)
+   */
+  gap?: number;
+  /**
+   * Space between the mark and the label (in px).
+   * @default theme.spacing(1)
+   */
+  markGap?: number;
```

Comment:
I think you can remove those props.

They were useful because with SVG we need to compute each item position. But with HTML, a dev can easily do 

```
<ChartsLegend sx={{ "& .mui-chart-legend-item": { marginRight: 10 } }} />
```